### PR TITLE
express: bamtools lib handling

### PR DESCRIPTION
**Pull request by baberlevi**
similar to #7447 this needs to handle the change in bamtools lib location

### CODE REVIEW

**Comment by adamjstewart** [comment]
If you play your cards right, you might even be able to use `spec['bamtools'].libs` which will automatically locate the library for you.

File: var/spack/repos/builtin/packages/express/package.py [hunk 1]
```diff
@@ -53,10 +53,8 @@ def patch(self):
                                 'bamtools'].prefix.include.bamtools.api)
             edit = FileFilter('CMakeLists.txt')
             # really ugly way to escape the $ in CMakeLists.txt
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/express/package.py]: You can get rid of this comment now. In fact, if you use `string=True`, you won't need to escape it at all.

**Comment by baberlevi** [comment]
Using the libs property seems to work fine. bamtools 2.4.1 includes both shared and static, and libs seems to return the shared library, whereas 2.5.1 seems to just include static, but I don't think that really matters ?

**Comment by adamjstewart** [comment]
@baberlevi This shouldn't matter:
https://github.com/spack/spack/blob/develop/lib/spack/spack/spec.py#L745

**Pull request merged**